business_rules/engine.py

```diff
@@ -1,75 +1,138 @@
+import logging
+
 from .fields import FIELD_NO_INPUT
 
-async def run_all(rule_list,
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidRuleDefinition(Exception):
+    """Invalid rule"""
+
+
+def get_value(rule_list, defined_variables, defined_actions) -> dict:
+    """
+    Run Rules till first will be triggered and returns its actions results.
+    Exception will be raised if more than one action was executed by the
+    triggered rule.
+    :returns: value returned by executed action
+    """
+    rule_results = run_all(rule_list, defined_variables, defined_actions,
+                           stop_on_first_trigger=True)
+    if len(rule_results) == 0:
+        raise InvalidRuleDefinition(
+            'No rule executed or no action found in matching rule'
+        )
+    actions_result = rule_results[0]
+    if len(actions_result) != 1:
+        raise InvalidRuleDefinition(
+            f'Expected only one action to be executed. '
+            f'Executed {actions_result} actions')
+
+    return actions_result[_get_first_key_in_dictionary(actions_result)]
+
+
+def _get_first_key_in_dictionary(dictionary):
+    """Get first key of the dict"""
+    return list(dictionary)[0]
+
+
+def run_all(rule_list,
             defined_variables,
             defined_actions,
-            stop_on_first_trigger=False):
-
-    rule_was_triggered = False
+            stop_on_first_trigger=False) -> list:
+    """ Run all Rules and return the rules actions results
+    :returns:
+        rule_results(list): list of dictionaries. Each dictionary is a rule
+        actions' results
+    """
+    logger.debug("business-rules starting")
+    rule_results = []
     for rule in rule_list:
-        result = await run(rule, defined_variables, defined_actions)
-        if result:
-            rule_was_triggered = True
+        actions_results = run(rule, defined_variables, defined_actions)
+        if actions_results:
+            rule_results.append(actions_results)
```

Comment:
@butsyk 
now `run_all` returns list:
```
    [
        {
            action_name: action_return_value,
        }
    ]
```
I think it will be good to know which rule has been applied for the action, because we can have several rules applying same action.
What about this?
```
    [
        {
            'action': action_name,
            'action_result': action_return_value,
            'rule': rule,
        }
    ]
```

Reply:
Should be easy to add to existing structure.